socket.go

```diff
@@ -273,6 +275,12 @@ func (socket *mongoSocket) Release() {
 				recycle = false
 			}
 		}
+		if socket.expiryTime != nil && time.Now().After(*socket.expiryTime) {
```

Comment:
thats true with this logic app can use a connection that at the time of returning to recycle list wasnt expired but at the time of acquire could be expired.

I've moved the expire logic to expire code path. Also took out the expiry logic from release method. This would make code little simpler but we might be holding connection little longer (which I think should be okay)